File: FromSlides/Fortran/08_global_loop_atomic.f90
PROGRAM loop_decompose
  
  USE omp_lib
  IMPLICIT NONE
  INTEGER, PARAMETER :: max_its = 10000
  INTEGER :: i, its_global

  its_global = 0
!$OMP PARALLEL DO
  DO i = 1, max_its
!The atomic section means that operations in this section
!cannot be interrupted by other threads. Only certain
!mathematical operations can be performed in an atomic section
!$OMP ATOMIC
    its_global = its_global + 1
!$OMP END ATOMIC
  END DO
!$OMP END PARALLEL DO

PRINT '(A,I0,A)','Counter records ', its_global, ' iterations'

END PROGRAM loop_decompose

File: FromSlides/C/08_global_loop_atomic.c
/*THIS PROGRAM DOESN'T WORK PROPERLY*/

#include <stdio.h>
#include <omp.h>

#define MAX_ITS 10000

int main(){

  int its_global, i;
  
  its_global = 0;
#pragma omp parallel for
  for (i=0;i<MAX_ITS;++i){
/*The atomic section means that operations in this section
cannot be interrupted by other threads. Only certain
mathematical operations can be performed in an atomic section*/

#pragma omp atomic
    {
      its_global++;
    }
  }

  printf("Counter records %i iterations\n", its_global);
}
